### ISSUE TITLE
Can't add T31x sensors to the KH100 hub

### ISSUE THREAD

**Issue by ghervasze** [petretiandrea/home-assistant-tapo-p100#751]
I am trying to add some temp sensors, that is T310 and T315. Managed to add the hub via the custom HACS integration but no devices (temp and humidity sensors) to it. They are simply not discovered. They do however show in the Tapo app and work fine. I get no errors regarding authentication or anything else. They should work as they are supported per the integration documentation. I appreciate any help.
Using the newest version of the integration. Attaching log and some screens.

I am new to all this so if anything is missing, kindly tell me how to get it and I will send it if needed.

[home-assistant_2024-04-16T16-02-48.791Z.log](https://github.com/petretiandrea/home-assistant-tapo-p100/files/14999571/home-assistant_2024-04-16T16-02-48.791Z.log)

![devices](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/167068000/23ea7389-10b4-477e-aeac-8c0bd0500a49)
![hub config](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/167068000/638777d1-b9b7-4d06-b656-f5c87ee221d2)
![integrations](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/167068000/9b566d22-6dbf-4c03-9cb7-b767001aae49)

**Comment by Gruffnco** [comment]
I have a similar problem I have the kh100 hub with 6 radiator valves which worked great in home assistant but since yesterday when the tapo hacs updated I now only get the kh100 device with 3 entities and no rad valves.
Please help.

**Comment by iambenmitchell** [comment]
Yep same problem, all 6 of my KE100 devices have become broken in the new update. 
<img width="794" alt="image" src="https://github.com/petretiandrea/home-assistant-tapo-p100/assets/27426792/f72e3d03-67de-4c18-8208-2506b9a28553">

**Comment by Seliaks65** [comment]
@Gruffnco @iambenmitchell - I found that the recent updates to the HACS Tapo integration killed my T310 temp sensors off and TRVs and made them unavailable. In the end, the only way I found to fix it was to redownload version 3.0.0 using HACs and restart Home Assistant and the sensors and TRVs came back. Not sure what's going on there with the 3.1.x versions. Info on how to redownload older packages is here [https://community.home-assistant.io/t/how-do-i-down-grade-a-hacs-loaded-custom-integration/683859](https://github.com/petretiandrea/home-assistant-tapo-p100/discussions/url)

**Comment by Seliaks65** [comment]
@ThiSchmidt - looks like you'll have to roll back to v3.0.0 to get those entities back.

**Comment by ghervasze** [comment]
I think it's safe to say this is a known issue and I expect it will be approached soon. That said, let's see how soon. Myself, I am at the moment waiting for a fix rather than roll back to 3.0.0. Hopefully it will come soon.

Another work around I red on the forums about, is to add the hub via matter. Apparently this method works well. If interested you can google for it. But I feel adding a separate integration for each device (from the same manufacturer) unnecessarily complicates the HA config and makes it harder to troubleshoot.

Strangely enough, I also lost a P100 plug on the way - to the now popular invalid auth error. But even stranger is that it's twin sibling (bought them in a 2 pack) works flawlessly. Obviously same fw, batch and all that. How weird is that?

**Comment by cmilanf** [comment]
I have the H100 and while initially I faced the same problem as commented here, it was solved when I added
```
tapo:
  discovery: true
```
in `configuration.yaml` file and fully restarted Home Assistant.

**Comment by Seliaks65** [comment]
@cmilanf that workaround didn't work for me

**Comment by laxdog** [comment]
Downgrading worked for me. Thanks.

Link above was broken for instructions, but I found them [here](https://community.home-assistant.io/t/how-do-i-down-grade-a-hacs-loaded-custom-integration/683859)

**Comment by iambenmitchell** [comment]
Is this still an issue? downgrading is such a hassle I haven't bothered upgrading to find out

**Comment by ThiSchmidt** [comment]
Yes, it is still an issue. It is also mentioned in #789 
I tried it for myself today again. Installed 3.1.4, restarted HA and only 3 entities in den KH100 hub are still there (firmware, signal level, overheat). I delete the integration totally and reinstalled 3.1.4, config all my Tapo devices new, same again. I downgraded to 3.0.0 and all entities in the KH100 hub are back. But only max 12 devices as mentioned in #650
